pandas/core/ops.py

```diff
@@ -1058,69 +1058,76 @@ def _bool_method_SERIES(cls, op, special):
     Wrapper function for Series arithmetic operations, to avoid
     code duplication.
     """
+    name = _get_op_name(op, special)
 
     def na_op(x, y):
         try:
             result = op(x, y)
         except TypeError:
-            if isinstance(y, list):
-                y = construct_1d_object_array_from_listlike(y)
-
-            if isinstance(y, (np.ndarray, ABCSeries)):
-                if (is_bool_dtype(x.dtype) and is_bool_dtype(y.dtype)):
-                    result = op(x, y)  # when would this be hit?
-                else:
-                    x = _ensure_object(x)
-                    y = _ensure_object(y)
-                    result = lib.vec_binop(x, y, op)
+            assert not isinstance(y, (list, ABCSeries, pd.Index))
+            if isinstance(y, np.ndarray):
+                # This next assertion is here because there used to be
```

Comment:
Yah, not sure what to put here.  ATM there is a non-reachable branch with a TODO comment specifically asking how it would be reached.  This seemed worth documenting somehow since someone at some point found it worth special-casing.

Reply:
i think the assertion pretty much explains this, though maybe add some text that we are trying to compare non-bool things here